distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1518,6 +1519,47 @@ public void testRetrievalReliabilityFails() throws Exception {
                 "FAILED");
     }
 
+    @Test
+    public void testMetacardCache() throws Exception {
+        String srcRequest = "{\"src\":\"" + OPENSEARCH_SOURCE_ID
+                + "\",\"start\":1,\"count\":250,\"cql\":\"anyText ILIKE '*'\",\"sort\":\"modified:desc\"}";
+
+        String cacheRequest =
+                "{\"src\":\"cache\",\"start\":1,\"count\":250,\"cql\":\"((anyText ILIKE '*') AND ((\\\"metacard_source\\\" = '"
+                        + OPENSEARCH_SOURCE_ID + "')))\",\"sort\":\"modified:desc\"}";
+
+        //This query will put the ingested metacards from the BeforeExam method into the cache
+        String cqlUrl = SEARCH_ROOT + "/catalog/internal/cql";
+        given().contentType("application/json")
+                .auth()
+                .basic(LOCALHOST_USERNAME, LOCALHOST_PASSWORD)
+                .body(srcRequest)
+                .when()
+                .post(cqlUrl)
+                .then()
+                .log()
+                .all()
+                .statusCode(200);
+
+        //CacheBulkProcessor could take up to 10 seconds to flush the cached results into solr
+        sleep(12000);
```

Comment:
Used console commands check the cache. Will wait up to 20 seconds for the cache to be populated.